### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by jp98** [AzureAD/microsoft-authentication-library-for-dotnet#3820]
**Logs and network traces**
Without logs or traces, it is unlikely that the team can investigate your issue. Capturing logs and network traces is described in [Logging wiki](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/logging).

**Which version of MSAL.NET are you using?**

**Platform**
.NET Framework 4.8

**What authentication flow has the issue?**
* Desktop 
    * [ ] Interactive

Other?

**Is this a new or existing app?**

b. The app is in production, I haven't upgraded MSAL, but started seeing this issue.

**Repro**

```csharp
	try
	{
		response = new AuthenticationResponse(await publicApp.AcquireTokenInteractive(registeredApp.Scopes).WithUseEmbeddedWebView(useEmbeddedWebView: true).ExecuteAsync());
	}
	catch (Exception ex)
	{
		response.Errors.Add(ex.Message ?? "");
	}
```

**Expected behavior**
Embedded browser prompts for authentication, and provides support for MFA.

**Actual behavior**
Browser prompts as expected, but then prompts with more information required. This in itself is ok, easy enough to step through in normal circumstances and provide additional information. For context, this issue does only happen if additional info is required. If no additional info is required, then it proceeds normally.

![image](https://user-images.githubusercontent.com/15373335/203421409-5ff04249-1625-4b97-a561-866ffd61cc93.png)

When clicking next, it thinks about it for a bit and then presents this screen.

![image](https://user-images.githubusercontent.com/15373335/203421568-46c5bb1a-d9de-4efc-9795-ffe071101b9a.png)

And authentication is not able to be completed

**Comment by jmprieur** [comment]
@jp98 
is your application using an embedded web view?

**Comment by jp98** [comment]
Yes it is using embedded web view as shown in the code provided.

**Comment by jmprieur** [comment]
I think that the embedded web view can't access the device state (which should be a device managed by your organization)
We suggest you use WAM (See https://learn.microsoft.com/en-us/azure/active-directory/develop/scenario-desktop-acquire-token-wam), or otherwise an embedded web view ()

**Comment by jmprieur** [comment]
Having said that the message from Azure AD is not explicit enough.

**Comment by jp98** [comment]
Essentially it takes you to confirm your details on the mysignins page - if you login to the account just using the standard browser and then complete that step and then come back to the app and login again, it works ok. But it's obviously inconvenient, and also weird that it suddenly decides at that point that the browser isn't compatible, even though it's taken you through the first steps successfully.

**Comment by bgavrilMS** [comment]
I confirm, the embedded webview is based on old IE. There are 2 ways around this: 

1. Use WAM. See https://aka.ms/msal-net-wam (please use the new preview implemenatation). This will require small app changes and registering a new redirect uri
2. Deploy a registry change to force the embedded webview to use Edge instead of IE

See details here:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/troubleshooting#javascript-errors-in-the-embedded-browser
